src/functions.jl

```diff
@@ -393,16 +421,11 @@ for T in (:Taylor1, :TaylorN)
                 return nothing
             end
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds c[k] = (k-1) * a[1] * c[k-1]
-            else
-                @inbounds mul!(c[k], (k-1)*a[1], c[k-1])
-            end
-            @inbounds for i = 2:k-1
+            @inbounds for i = 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * a[i] * c[k-i]
                 else
-                    mul!(c[k], (k-i)*a[i], c[k-i])
+                    mul!(c[k], a[i], c[k-i], scalar=k-i)
                 end
             end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done.